### PR TITLE
Fix/tca 389/fix s3 latency issue during item import

### PR DESCRIPTION
**Pull request by SashaTAO**
JIRA: https://oat-sa.atlassian.net/browse/TCA-389

Problem:
When using S3 flysystem adapter during test import if we call `getSIze()` or `getMimeType()` of just imported file it randomly results in FileNotFound exception (due to latency issue, files appears in S3 after that).

Solution:
After file import use local file source (which was just uploaded to S3) to get file metadata.

How to test: 
- Import QTI item and check that it renders correctly including all required media files.

### CODE REVIEW

**Comment by SashaTAO** [comment]
> I like your solution. Since we have an asynchronous write, we can't act on the destination file directly. So using the source file is a good way to avoid the issue.
> The only problem that we can still have is that we can't be sure that the file actually is written to s3...

We rely on Flysystem here. It returns boolean value as result of file upload. If it's `true` we can assume that file was uploaded successfully.

**Comment by julien-sebire** [comment]
You've got my approval on this! :)

File: test/unit/models/classes/ItemsServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
```
[the remark this answers is not in the document]
Comment (SashaTAO) [reply test/unit/models/classes/ItemsServiceTest.php:19]: Done

File: test/unit/models/classes/ItemsServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+namespace oat\taoItems\test\unit\models\classes;
+
+use taoItems_models_classes_ItemsService;
+use oat\generis\test\TestCase;
+
+class ItemsServiceTest extends TestCase
+{
+    public function testComposeItemDirectoryPathReturnsStringPath()
+    {
+        $object = new taoItems_models_classes_ItemsService();
```
[the remark this answers is not in the document]
Comment (SashaTAO) [reply test/unit/models/classes/ItemsServiceTest.php:29]: It's just personal preference. I prefer to use this name for tested class in all unit tests. 
Renamed.

File: test/unit/models/classes/ItemsServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        $itemDirectoryName = 'FAKE_DIRECTORY_NAME';
+        $language = 'FAKE_LANGUAGE';
+
+        $result = $object->composeItemDirectoryPath($itemDirectoryName, $language);
```
[the remark this answers is not in the document]
Comment (SashaTAO) [reply test/unit/models/classes/ItemsServiceTest.php]: Done

**Comment by zagovorichev** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**